### ISSUE TITLE
H100: Error "ms_deformable_im2col_cuda"

### ISSUE THREAD

**Issue by haha-yuki-haha** [yifan123/reward-server#9]
Package                  Version      Editable project location
------------------------ ------------ ---------------------------------------------------------------
addict                   2.4.0
aliyun-python-sdk-core   2.16.0
aliyun-python-sdk-kms    2.16.5
blinker                  1.9.0
braceexpand              0.1.7
certifi                  2025.11.12
cffi                     2.0.0
charset-normalizer       3.4.4
click                    8.3.1
clip-benchmark           1.6.1
colorama                 0.4.6
contourpy                1.3.2
crcmod                   1.7
cryptography             46.0.3
cycler                   0.12.1
filelock                 3.14.0
Flask                    3.1.0
fonttools                4.60.1
fsspec                   2025.10.0
ftfy                     6.3.1
gunicorn                 23.0.0
hf-xet                   1.2.1
huggingface-hub          0.36.0
idna                     3.11
itsdangerous             2.2.0
Jinja2                   3.1.6
jmespath                 0.10.0
joblib                   1.5.2
kiwisolver               1.4.9
Markdown                 3.10
markdown-it-py           4.0.0
MarkupSafe               3.0.3
matplotlib               3.10.7
mdurl                    0.1.2
mmcv-full                1.7.2
mmdet                    2.27.0       /XXXX/mmdetection-2.27.0
mmengine                 0.10.7
model-index              0.1.11
mpmath                   1.3.0
networkx                 3.4.2
numpy                    1.26.0
nvidia-cublas-cu12       12.4.5.8
nvidia-cuda-cupti-cu12   12.4.127
nvidia-cuda-nvrtc-cu12   12.4.127
nvidia-cuda-runtime-cu12 12.4.127
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.1.3
nvidia-curand-cu12       10.3.5.147
nvidia-cusolver-cu12     11.6.1.9
nvidia-cusparse-cu12     12.3.1.170
nvidia-cusparselt-cu12   0.6.2
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.4.127
open_clip_torch          2.31.0
opencv-python            4.11.0.86
opendatalab              0.0.10
openmim                  0.3.9
openxlab                 0.1.3
ordered-set              4.1.0
oss2                     2.17.0
packaging                24.2
pandas                   2.3.3
pillow                   12.0.0
pip                      25.3
platformdirs             4.5.0
pycocoevalcap            1.2
pycocotools              2.0.10
pycparser                2.23
pycryptodome             3.23.0
Pygments                 2.19.2
pyparsing                3.2.5
python-dateutil          2.9.0.post0
pytz                     2023.4
PyYAML                   6.0.3
regex                    2025.11.3
requests                 2.28.2
rich                     13.4.2
safetensors              0.7.0
scikit-learn             1.7.2
scipy                    1.15.3
setuptools               60.2.0
shapely                  2.1.2
six                      1.17.0
sympy                    1.13.1
tabulate                 0.9.0
termcolor                3.2.0
terminaltables           3.1.10
threadpoolctl            3.6.0
timm                     1.0.22
tokenizers               0.21.4
tomli                    2.3.0
torch                    2.1.2+cu121
torchvision              0.16.2+cu121
tqdm                     4.65.2
transformers             4.51.3
triton                   2.1.0
typing_extensions        4.15.0
tzdata                   2025.2
urllib3                  1.26.20
wcwidth                  0.2.14
webdataset               1.0.2
Werkzeug                 3.1.3
wheel                    0.45.1
yapf                     0.43.0, 
------------------------ ------------ ---------------------------------------------------------------
Hello, H100 makes a error "error in ms_deformable_im2col_cuda: no kernel image is available for execution on the device" while I use your method "mim uninstall mmcv-full 
TORCH_CUDA_ARCH_LIST="9.0" pip install mmcv-full" 
Could you please how to solve this?Thanks

**Comment by haha-yuki-haha** [comment]
> Package Version Editable project location
> 
> addict 2.4.0 aliyun-python-sdk-core 2.16.0 aliyun-python-sdk-kms 2.16.5 blinker 1.9.0 braceexpand 0.1.7 certifi 2025.11.12 cffi 2.0.0 charset-normalizer 3.4.4 click 8.3.1 clip-benchmark 1.6.1 colorama 0.4.6 contourpy 1.3.2 crcmod 1.7 cryptography 46.0.3 cycler 0.12.1 filelock 3.14.0 Flask 3.1.0 fonttools 4.60.1 fsspec 2025.10.0 ftfy 6.3.1 gunicorn 23.0.0 hf-xet 1.2.1 huggingface-hub 0.36.0 idna 3.11 itsdangerous 2.2.0 Jinja2 3.1.6 jmespath 0.10.0 joblib 1.5.2 kiwisolver 1.4.9 Markdown 3.10 markdown-it-py 4.0.0 MarkupSafe 3.0.3 matplotlib 3.10.7 mdurl 0.1.2 mmcv-full 1.7.2 mmdet 2.27.0 /XXXX/mmdetection-2.27.0 mmengine 0.10.7 model-index 0.1.11 mpmath 1.3.0 networkx 3.4.2 numpy 1.26.0 nvidia-cublas-cu12 12.4.5.8 nvidia-cuda-cupti-cu12 12.4.127 nvidia-cuda-nvrtc-cu12 12.4.127 nvidia-cuda-runtime-cu12 12.4.127 nvidia-cudnn-cu12 9.1.0.70 nvidia-cufft-cu12 11.2.1.3 nvidia-curand-cu12 10.3.5.147 nvidia-cusolver-cu12 11.6.1.9 nvidia-cusparse-cu12 12.3.1.170 nvidia-cusparselt-cu12 0.6.2 nvidia-nccl-cu12 2.21.5 nvidia-nvjitlink-cu12 12.4.127 nvidia-nvtx-cu12 12.4.127 open_clip_torch 2.31.0 opencv-python 4.11.0.86 opendatalab 0.0.10 openmim 0.3.9 openxlab 0.1.3 ordered-set 4.1.0 oss2 2.17.0 packaging 24.2 pandas 2.3.3 pillow 12.0.0 pip 25.3 platformdirs 4.5.0 pycocoevalcap 1.2 pycocotools 2.0.10 pycparser 2.23 pycryptodome 3.23.0 Pygments 2.19.2 pyparsing 3.2.5 python-dateutil 2.9.0.post0 pytz 2023.4 PyYAML 6.0.3 regex 2025.11.3 requests 2.28.2 rich 13.4.2 safetensors 0.7.0 scikit-learn 1.7.2 scipy 1.15.3 setuptools 60.2.0 shapely 2.1.2 six 1.17.0 sympy 1.13.1 tabulate 0.9.0 termcolor 3.2.0 terminaltables 3.1.10 threadpoolctl 3.6.0 timm 1.0.22 tokenizers 0.21.4 tomli 2.3.0 torch 2.1.2+cu121 torchvision 0.16.2+cu121 tqdm 4.65.2 transformers 4.51.3 triton 2.1.0 typing_extensions 4.15.0 tzdata 2025.2 urllib3 1.26.20 wcwidth 0.2.14 webdataset 1.0.2 Werkzeug 3.1.3 wheel 0.45.1 yapf 0.43.0,
> 
> Hello, H100 makes a error "error in ms_deformable_im2col_cuda: no kernel image is available for execution on the device" while I use your method "mim uninstall mmcv-full TORCH_CUDA_ARCH_LIST="9.0" pip install mmcv-full" Could you please how to solve this?Thanks

Thanks, I solve this problem as follow: (python=3.10)
------------------------ ------------ ---------------------------------------------------------------
pip install torch-2.1.2+cu121-cp310-cp310-linux_x86_64.whl
pip install torchvision-0.16.2+cu121-cp310-cp310-linux_x86_64.whl
pip install gunicorn==23.0.0 openmim==0.3.9 open-clip-torch==2.31.0 numpy==1.26.0 opencv-python==4.11.0.86 clip-benchmark==1.6.1 flask==3.1.0

unzip mmcv-1.7.2.zip
cd mmcv-1.7.2
pip install mmengine openmim
pip install "setuptools>=69.0.3"
(For the setup.py, we should set EXT_TYPE = 'pytorch' in line 26 before we pip install)
MMCV_WITH_OPS=1 MMCV_CUDA_ARGS="-arch=sm_90" pip install -v -e . --no-build-isolation

cd ..
unzip mmdetection-2.27.0.zip
cd mmdetection-2.27.0
MMCV_CUDA_ARGS="-arch=sm_90" pip install -v -e . --no-build-isolation
pip install transformers==4.51.1 
------------------------ ------------ ---------------------------------------------------------------

**Comment by beautyremain** [comment]
hi, I encounter the same issue on H100 80G.  May I ask where can I get the mmcv-1.7.2.zip and mmdetection-2.27.0.zip. Is it from the official?

**Comment by haha-yuki-haha** [comment]
> hi, I encounter the same issue on H100 80G. May I ask where can I get the mmcv-1.7.2.zip and mmdetection-2.27.0.zip. Is it from the official?

Yes.
mmcv-1.7.2: https://github.com/open-mmlab/mmcv/tree/v1.7.2
mmdetection-2.27.0: https://github.com/open-mmlab/mmdetection/tree/v2.27.0

**Comment by beautyremain** [comment]
Thank you so much! It has bothered me for several days, the author's method didn't work for me, and it finally been solved exactly according to your method.
